src/main/java/ru/rt/restream/reindexer/convert/FieldConverterRegistryFactory.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2020 Restream
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ru.rt.restream.reindexer.convert;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.rt.restream.reindexer.annotations.Convert;
+import ru.rt.restream.reindexer.convert.util.ConversionUtils;
+import ru.rt.restream.reindexer.convert.util.ResolvableType;
+import ru.rt.restream.reindexer.util.Pair;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * For internal use only, as this contract is likely to change.
+ */
+public enum FieldConverterRegistryFactory implements FieldConverterRegistry {
+    INSTANCE;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FieldConverterRegistryFactory.class);
+
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+    private final Map<Class<?>, GenericFieldConverter<?, ?>> converters = new HashMap<>();
+
+    private final Map<ResolvableType, GenericFieldConverter<?, ?>> globalConverters = new HashMap<>();
```

Comment:
Without studying the code, it is not clear what the two fields with similar meanings are for - converters and globalConverters.

Reply:
`converters` stores the created converters within `@Convert` annotation, in case when two fields share the same `@Convert.converterClass` it should reuse the same instance.
`globalConverters` is the globally defined converts by the user or an external library, like for example if we need to declare a converter for LocalDateTime, there should be no need to apply it for each field manually via `@Convert` or `registerFieldConverter`.